### ISSUE TITLE
[cygnus-ngsi][NameMapping] Allow deep Json attribute mapping.

### ISSUE THREAD

**Issue by pmo-sdr** [telefonicaid/fiware-cygnus#1856]
It will be nice to be able to map specific NGSI Json attributes to a final persisted field.

#### Example: 

We have a NGSI entity with address field like this:

```json
"address": {
    "type": "PostalAddress",
    "addressLocality": "Seattle",
    "addressRegion": "WA",
    "postalCode": "98052",
    "streetAddress": "20341 Whitworth Institute 405 N. Whitworth"
  }
```

But we want to have a Mysql table with fields,  `city` and `region`
To do this, namemapping config should look like:

```json
"attributeMappings": [
                        {
                           "originalAttributeName": "address.addressLocality",
                           "newAttributeName": "city",
                           ...
                        }, 
                        {
                           "originalAttributeName": "address.addressRegion",
                           "newAttributeName": "region",
                           ...
                        },
```
Mapping could use JsonPath-like expressions.

**Comment by pmo-sdr** [comment]
Solution "Highlights":  
- Mapping interceptor "knows" if provided expression is a regex or a Jsonpath expression.
- If JsonPath, original attribute name is extracted from JsonPath expression
- Mapping can be configured to non stop at first match.

**Comment by manucarrace** [comment]
It seems to be a interesting feature.
What do you mean with "Mapping can be configured not to stop at first match"? does it mean that when an attribute regexp is matched, it doesn't go on with the next attribute?

**Comment by pmo-sdr** [comment]
Yes, nowadays, name mapping stops checking mapping rules when it finds a match. 
It makes sense, since it's the most common behavior, and stopping it produces a significant performance improvement.

But sometimes, we may need to "clone" or "split" NGSI fit final persistence structure, this is why the new feature will be configurable at sink level.

**Comment by pmo-sdr** [comment]
New section added to docs explaining how it will work.  'Advancen Mapping' section, 'Clone attributes' and 'Using JsonPath Expressions'
https://github.com/Stdri/fiware-cygnus/blob/JsonPath/doc/cygnus-ngsi/installation_and_administration_guide/name_mappings.md

**Comment by manucarrace** [comment]
Nice. For me it's a very interesting feature but only for attributes. I see no point in apply it to entities, subservices or services, for the time being ...

**Comment by fgalan** [comment]
PR https://github.com/telefonicaid/fiware-cygnus/pull/1860
